addons/website/static/src/js/utils.js

```diff
@@ -33,25 +33,45 @@ function loadAnchors(url) {
  * @param {jQuery} $input
  */
 function autocompleteWithPages(self, $input) {
-    $input.autocomplete({
+    $.widget("website.urlcomplete", $.ui.autocomplete, {
+        _create: function () {
+            this._super();
+            this.widget().menu("option", "items", "> :not(.ui-autocomplete-category)");
+        },
+        _renderMenu: function (ul, items) {
+            const self = this;
+            for (let [key, value] of Object.entries(items['0'])) {
+                if (key !== 'matching_pages' && value.length) {
+                    key = key.replace(/_/g, ' ');
+                    key = key.charAt(0).toUpperCase() + key.slice(1);
+                    ul.append("<li class='ui-autocomplete-category font-weight-bold text-center py-2'>" + key + "</li>");
+                }
+                value.forEach(suggestion => {
+                     self._renderItem(ul, suggestion);
+                });
+            }
+        },
+        _renderItem: function (ul, item) {
+            return $("<li>")
+                   .data('ui-autocomplete-item', item)
+                   .append(`<div>${item.label}</div>`)
+                   .appendTo(ul);
+        }
+    });
+    $input.urlcomplete({
         source: function (request, response) {
             if (request.term[0] === '#') {
                 loadAnchors(request.term).then(function (anchors) {
                     response(anchors);
                 });
             } else {
                 return self._rpc({
```

Comment:
jsonrpc directly

Reply:
@DramixDW Here you should still use `self._rpc`, but JKE told me you had issues with it? Check with me tomorrow directly.